tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingWildcardNameTests.java

```diff
@@ -0,0 +1,138 @@
+import java.util.Map;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.opentracing.tck.application.WildcardClassService;
+import org.eclipse.microprofile.opentracing.tck.tracer.TestSpan;
+import org.eclipse.microprofile.opentracing.tck.tracer.TestSpanTree;
+import org.eclipse.microprofile.opentracing.tck.tracer.TestSpanTree.TreeNode;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public class OpenTracingWildcardNameTests extends OpenTracingClientBaseTests {
+
+    public static class TestConfiguration implements ConfigSource {
+        private Map<String, String> propMap = new HashMap<>();
+
+        {
+            propMap.put("mp.opentracing.server-operation-name-provider", "http-path");
+        }
+
+        @Override
+        public Map<String, String> getProperties() {
+            return propMap;
+        }
+
+        @Override
+        public String getValue(String s) {
+            return propMap.get(s);
+        }
+
+        @Override
+        public String getName() {
+            return this.getClass().getName();
+        }
+    }
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return OpenTracingBaseTests.createDeployment()
+            .addAsServiceProvider(ConfigSource.class, TestConfiguration.class);
+    }
+
+    @Override
+    protected String getOperationName(String spanKind, String httpMethod, Class<?> clazz, Method method) {
+        if (spanKind.equals(Tags.SPAN_KIND_SERVER)) {
+            StringBuilder operationName = new StringBuilder();
+            Path classPath = clazz.getAnnotation(Path.class);
+            if (classPath != null) {
+                if (!classPath.value().startsWith("/")) {
+                    operationName.append("/");
+                }
+                debug(classPath.value());
+                operationName.append(classPath.value());
+            }
+            if (!classPath.value().endsWith("/")) {
+                operationName.append("/");
+            }
+            Path methodPath = method.getAnnotation(Path.class);
+            String methodPathStr = methodPath.value();
+            debug(methodPathStr);
+            if (methodPathStr.startsWith("/")) {
+                methodPathStr = methodPathStr.replaceFirst("/", "");
+            }
+            operationName.append(methodPathStr);
+            return operationName.toString();
+        }
+        return super.getOperationName(spanKind, httpMethod, clazz, method);
+    }
+
+    /**
+     * Test that server endpoint is adding standard tags
+     */
+    @Test
+    @RunAsClient
+    private void testWildcard() throws InterruptedException {
```

Comment:
Should also have a test with just a `Path` on the method.

Reply:
Is it valid in jax-rs?

Reply:
After a quick check, can't find anywhere where it shows it only on the method, although thought I had seen an example before. So don't worry.

Reply:
It's a valid point, I will also investigate.

Reply:
https://stackoverflow.com/a/45081184 It says path on the class cannot be omitted.